validate.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+RUNS_PER_THREAD=100000
+MAX_PTHREADS=$(nproc)
+BIN_PATH="./build/bin"
+RES_FILE="res.txt"
+BUILD_LOG="build.log"
+STEP_SELETCTED=0
+STEP_THREADS=1
+FIBERS=""
+NUMA_NODES=""
+PASS_STATUS=1
+
+function usage()
+{
+    echo -e "Usage: ./validate.sh OPTION=NUM ...";
+    echo -e "This script compiles the sources in DEBUG mode and validates the correctnes of some of the provided concurrent objects.";
+    echo -e ""
+    echo -e "The following options are available."
+    echo -e "-t, --max_threads \t set the maximum number number of POSIX threads to be used in the last set of iterations of the benchmark, default is the number of system_cpus"
+    echo -e "-s, --step \t set the step (extra number of POSIX threads to be used) in succesive set of iterations of the benchmark, default is number of system_cpu/8 or 1"
+    echo -e "-f, --fibers  \t set the number of fibers (user-level threads) per posix thread."
+    echo -e "-r, --runs    \t set the total number of operations executed by each thread of each benchmark, default is ${RUNS_PER_THREAD}"
+    echo -e "-w, --max_work\t set the amount of workload (i.e. dummy loop iterations among two consecutive operations of the benchmarked object), default is 64"
+    echo -e ""
+    echo -e "-h, --help    \t displays this help and exits"
+    echo -e ""
+}
+
+COLOR_PASS="[ \e[32mPASS\e[39m ]"
+COLOR_FAIL="[ \e[31mFAIL\e[39m ]"
+
+declare -a uobjects=("ccsynchbench.run"                     "dsmsynchbench.run" "hsynchbench.run" "oscibench.run"      "simbench.run")
+declare -a queues=(  "ccqueuebench.run" "clhqueuebench.run" "dsmqueuebench.run" "hqueuebench.run" "osciqueuebench.run" "simqueuebench.run")
+declare -a stacks=(  "ccstackbench.run" "clhstackbench.run" "dsmstackbench.run" "hstackbench.run" "oscistackbench.run" "simstackbench.run")
+
+declare -a validate_objects=()
+
+validate_objects+=${uobjects[@]}
+validate_objects+=${queues[@]}
+validate_objects+=${stacks[@]}
+
+if [ "$1" = "--help" ] || [ "$1" = "-h" ]; then
+    usage;
+    exit;
+fi
+
+while [ "$1" != "" ]; do
+    PARAM=`echo $1 | awk -F= '{print $1}'`
+    VALUE=`echo $1 | awk -F= '{print $2}'`
+    case $PARAM in
+        -h | --help)
+            usage;
+            exit;
+            ;;
+        -s | --step_threads)
+            STEP_THREADS=$VALUE
+            STEP_SELETCTED=1
+            ;;
+        -t | --max_threads)
+            MAX_PTHREADS=$VALUE
+            ;;
+        -f | --fibers)
+            FIBERS="-f $VALUE"
+            ;;
+        -w | --max_work)
+            WORKLOAD="-w $VALUE"
+            ;;
+        -n | --numa_nodes)
```

Comment:
Either mention it in the help, or remove this code here if the value for the test is always going to be `""`

Reply:
See comments above.

Reply:
This option is not mentioned in the help section.

Reply:
OK, This seems to be missing also in the bench.sh.
It is documented in the README file, but is missing in bench.sh and validate.sh.
I'm adding this.